### ISSUE TITLE
Warning when using value and radio component

### ISSUE THREAD

**Issue by maciejmyslinski** [final-form/react-final-form#112]
### Are you submitting a **bug report** or a **feature request**?
bug report

### What is the current behavior?
[![Edit 0xr45vmryl](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/0xr45vmryl)
There is a warning in the console while doing this:
```js
const Radio = ({ input, label }) => (
  <label>
    <input type="radio" {...input} />
    {label}
  </label>
);

const App = () => (
  <Form
    onSubmit={() => {}}
    render={() => (
      <Field
        name="example"
        type="radio"
        value="yes"
        label="Yes"
        component={Radio}
      />
    )}
  />
);
```

### What is the expected behavior?
I should be able to set the value for `radio` component rendered with `component` prop.

**Comment by erikras** [comment]
Hmm... That warning was meant to be helpful, but your usage (i.e. a custom `Radio` component) looks totally reasonable, so perhaps it should be removed...

**Comment by maciejmyslinski** [comment]
I appreciate your efforts to warn a user about improper usage 👍

**Comment by erikras** [comment]
Published fix in [`v3.0.2`](https://github.com/final-form/react-final-form/releases/tag/v3.0.2).

**Comment by maciejmyslinski** [comment]
I'm amazed how you maintain this lib! 🎉 👏
